src/app/public/testing/sort/sort-fixture.ts

```diff
@@ -0,0 +1,151 @@
+import {
+  DebugElement
+} from '@angular/core';
+
+import {
+  ComponentFixture
+} from '@angular/core/testing';
+
+import {
+  By
+} from '@angular/platform-browser';
+
+import {
+  SkyAppTestUtility
+} from '@skyux-sdk/testing';
+
+import {
+  SkySortFixtureMenu
+} from './sort-fixture-menu';
+
+import {
+  SkySortFixtureMenuItem
+} from './sort-fixture-menu-item';
+
+/**
+ * Provides information for and interaction with a SKY UX sort component.
+ * By using the fixture API, a test insulates itself against updates to the internals
+ * of a component, such as changing its DOM structure.
+ */
+export class SkySortFixture {
+
+  /**
+   * The active sort menu item, if one exists. Menu items are only available when the menu dropdown
+   * is open. If the menu dropdown is closed, this property will be undefined.
+   */
+  public get activeMenuItem(): SkySortFixtureMenuItem {
+    return this.menuItems?.find(x => x.isActive);
+  }
+
+  /**
+   * The sort menu's properties.
+   */
+  public get menu(): SkySortFixtureMenu {
+    return {
+      buttonText: SkyAppTestUtility.getText(this.getSortButtonTextSpan()),
+      isOpen: this.getDropdownMenuEl() !== null
+    };
+  }
+
+  /**
+   * The properties of each sort menu item. Menu items are only available when the menu dropdown
+   * is open. If the menu dropdown is closed, this property will be undefined.
+   */
+  public get menuItems(): SkySortFixtureMenuItem[] {
+    // Return undefined when we can't determine what the options are.
+    // We do this to avoid any confusion with an empty set of options.
+    if (!this.menu.isOpen) {
+      return undefined;
+    }
+
+    return this.getSortItems()
+      .map((item: HTMLElement) => {
+        const itemButton = item.querySelector('button');
+
+        return {
+          isActive: item.classList.contains('sky-sort-item-selected'),
+          text: SkyAppTestUtility.getText(itemButton)
+        };
+      });
+  }
+
+  private _debugEl: DebugElement;
+
+  constructor(
+    private fixture: ComponentFixture<any>,
+    skyTestId: string
+  ) {
+    this._debugEl = SkyAppTestUtility.getDebugElementByTestId(fixture, skyTestId, 'sky-sort');
+  }
+
+  /**
+   * Ensures the sort menu is open and selects the menu item with the specified text,
+   * if a matching item is available.
+   * @param menuItemText The text of the menu item to select.
+   */
+  public async selectMenuItem(menuItemText: string): Promise<void> {
```

Comment:
I would rename this to `selectMenuItemByText` (or something similar) to be more specific, in case we need to select an item by another means in the future (such as by its index).

Along these same lines, what's the reasoning behind selecting an item by text, instead of by index?